### PR TITLE
Feature: able to use argument/flag to accept project directory path

### PR DESCRIPTION
**Pull request by dzakirin-muhammad**
#15 
🔥🔥🔥

### CODE REVIEW

**Review by zulkhairi-edag — APPROVED** [review]
LGTM

File: example/build.gradle [hunk 1]
```diff
@@ -1,23 +1,30 @@
 plugins {
```
Comment (soon-yoong) [inline example/build.gradle:1]: changes done to this file to downgrade the requirement from java 17 to java 1.8

File: adapters/gradle/gradle.go [hunk 2]
```diff
@@ -14,30 +13,33 @@ type Gradle struct {
...
+		log.Println("Error injecting cycloneDx plugin to build.gradle, ", injectErr)
 	}
 
-	workingDir := "SBOMWorkingDir/" + resultInfo.Uuid + "/"
-
 	var shell = structs.NewShell(workingDir)
 
 	var gradleArgument = []string{
 		"cyclonedxBom",
-		"--build-file",
```
Comment (soon-yoong) [inline adapters/gradle/gradle.go:30]: as the Shell cmd already changing directory to the target folder, specifying build-file is no longer needed

File: structs/Shell.go [hunk 3]
```diff
@@ -14,22 +13,16 @@ type Shell struct {
 	Warnings []string `json:"warnings"`
 }
 
-func NewShell(folder string) Shell {
+func NewShell(workingDir string) Shell {
```
Comment (soon-yoong) [inline structs/Shell.go:16]: Changed to accept a full workingDir path instead of just a folder name

**Review by soon-yoong — APPROVED** [review]

File: structs/Filesystem.go [hunk 4]
```diff
@@ -48,7 +48,7 @@ func (filesystem *Filesystem) Scan() []ResultInfo {
 
 	var result []ResultInfo
 
-	err := filepath.WalkDir(".", func(path string, info fs.DirEntry, err error) error {
+	err := filepath.WalkDir(filesystem.Root, func(path string, info fs.DirEntry, err error) error {
```
Comment (soon-yoong) [inline structs/Filesystem.go:51]: existing bug fix, scan method should look for the directory set via Root not taking "."

**Pull request merged**